pkgs/servers/conduit/default.nix

```diff
@@ -0,0 +1,21 @@
+{ stdenv, lib, fetchFromGitLab, rustPlatform, }:
+rustPlatform.buildRustPackage rec {
+  pname = "conduit";
+  version = "git";
+
+  src = fetchFromGitLab {
+    owner = "famedly";
+    repo = pname;
+    rev = "1aec80c57c02ef17475436c530b2ac2bc947b923";
+    sha256 = "sha256-bz+afiRaGQ5wHRVzEXNoaFuS36JU9sOGjQ/7xQtl458=";
+  };
+  cargoSha256 = "sha256-lhoJNOWxwQ8LuRgv8clYWwyn/RS8mOGvfGp3yuvSj7I=";
+
```

Comment:
A previous package by @piegames used openssl, is that not the case anymore?

https://github.com/piegamesde/nur-packages/blob/master/pkgs/matrix-conduit.nix

Reply:
Good catch! That rust openssl bindings lib has the *nice* fallback to just compile their own openssl when they can't find an upstream one, leaving you with a hidden ticking time bomb in terms of vulnerabilities.

@piegamesde Did you verify that your additions are actually enough? I built conduit with
```
  nativeBuildInputs = [ pkgconfig ];
  buildInputs = [ openssl ];
```
but there is no hint of the openssl lib being used:
```
~ldd result/bin/conduit
        linux-vdso.so.1 (0x00007ffc0bfe4000)
        libgcc_s.so.1 => /nix/store/vaf92wc23m067q9aig6zjkcnjvrg768n-glibc-2.32-46/lib/libgcc_s.so.1 (0x00007f830e408000)
        libpthread.so.0 => /nix/store/vaf92wc23m067q9aig6zjkcnjvrg768n-glibc-2.32-46/lib/libpthread.so.0 (0x00007f830e3e7000)
        libm.so.6 => /nix/store/vaf92wc23m067q9aig6zjkcnjvrg768n-glibc-2.32-46/lib/libm.so.6 (0x00007f830e2a4000)
        libdl.so.2 => /nix/store/vaf92wc23m067q9aig6zjkcnjvrg768n-glibc-2.32-46/lib/libdl.so.2 (0x00007f830e29f000)
        libc.so.6 => /nix/store/vaf92wc23m067q9aig6zjkcnjvrg768n-glibc-2.32-46/lib/libc.so.6 (0x00007f830e0de000)
        /nix/store/vaf92wc23m067q9aig6zjkcnjvrg768n-glibc-2.32-46/lib/ld-linux-x86-64.so.2 => /nix/store/v8q6nxyppy1myi3rxni2080bv8s9jxiy-glibc-2.32-40/lib64/ld-linux-x86-64.so.2 (0x00007f830fbb6000)
```

Reply:
afaik conduit uses rustls and doesn't depend on openssl at all. the only mention of openssl i can find is the openssl-probe crate which i think is depended on by rustls for finding openssl certificates in the system

Reply:
Checked with upstream. No dependency on openssl is to be found.